library/src/main/java/com/bumptech/glide/load/resource/gif/GifDrawable.java

```diff
@@ -298,7 +315,47 @@ public void setAlpha(int i) {
 
   @Override
   public void setColorFilter(ColorFilter colorFilter) {
-    getPaint().setColorFilter(colorFilter);
+    if (getColorFilter() != colorFilter) {
```

Comment:
why use reference equality instead of Objects.equals?

and why does this specific method check for equality but the other methods not? I guess this is trying to handle the case where colorFilter is null to prevent it from clearing the tint? If that's the case please can you make the code explicit (preferred) or add a comment to say it is, because otherwise it's not at all obvious from the code that this is the intent.

Reply:
I added the check because GradientDrawable does the same thing.
However, I think it is not directly related to this PR, so please let me revert this change to simplify this PR.